csi/controllerserver.py

```diff
@@ -590,7 +596,42 @@ def ControllerExpandVolume(self, request, context):
             expand_volume(mntdir)
         else:
             update_subdir_volume(
-                mntdir, pvname, expansion_requested_pvsize)
+                mntdir, existing_volume.hostvoltype, pvname, expansion_requested_pvsize)
+            if existing_volume.hostvoltype == 'External':
+                # Use Gluster quota if set
+                if (os.path.isfile("/etc/secret-volume/ssh-privatekey") \
+                    and "SECRET_GLUSTERQUOTA_SSH_USERNAME" in os.environ):
+                    use_gluster_quota = True
+
+        # Can be true only if its 'External'
+        if use_gluster_quota:
+            secret_private_key = "/etc/secret-volume/ssh-privatekey"
+            secret_username = os.environ.get('SECRET_GLUSTERQUOTA_SSH_USERNAME', None)
+
+            logging.debug(logf("Set Quota (expand) using gluster directory Quota"))
+            quota_cmd = [
+                "ssh",
+                "-oStrictHostKeyChecking=no",
+                "-i",
+                "%s" % secret_private_key,
+                "%s@%s" % (secret_username, existing_volume.ghost),
+                "sudo",
+                "gluster",
+                "volume",
+                "quota",
+                "%s" % existing_volume.gvolname,
+                "limit-usage",
+                "/%s" % existing_volume.volpath,
+                "%s" % expansion_requested_pvsize * 0.05,
+            ]
+            try:
+                execute(*quota_cmd)
+            except CommandException as err:
+                errmsg = "Unable to set Gluster Quota via ssh"
+                logging.error(logf(errmsg, error=err))
+                context.set_details(errmsg)
+                context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+                return csi_pb2.CreateVolumeResponse()
```

Comment:
return csi_pb2.ControllerExpandVolumeResponse()

Reply:
ig, it's better to have an `else` as well, like some users may not use gluster quota and in that case exapandvol should just log a msg?